devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -502,11 +510,15 @@ private void handleAutoCompile() throws MojoExecutionException {
 
             if (POST_COMPILE_PHASES.contains(goal)) {
                 compileNeeded = false;
-                break;
             }
+
+            if (jandexGoalPhase != null
+                    && POST_COMPILE_PHASES.indexOf(goal) >= POST_COMPILE_PHASES.indexOf(jandexGoalPhase)) {
```

Comment:
I changed this (in a spearate commit) so that the plugin is not executed when a phase higher is executed (e.g. process-test-classes).
That's also why `Set` became `List`.